apps/web/src/app/not-found.tsx

```diff
@@ -0,0 +1,75 @@
+import Image from 'next/image';
+import { Button } from '@repo/ui/components/button';
+import { getRandomChallenge } from '~/utils/server/get-random-challenge';
+import { getScopedI18n } from '~/locales/server';
+import { setStaticParamsLocale } from 'next-international/server';
+
+export default async function NotFound() {
+  // fix broken e2e tests... not sure why this is needed
+  // but i18n still works with other locales even with this set
+  setStaticParamsLocale('en');
```

Comment:
actually just discovered this is not true.... stand by, looking at how to fix.

Reply:
@QuiiBz I'm getting the following error only during pnpm build (pnpm dev works great)... when I add the code above I can supress the error and build succeeds, but breaks i18n on the 404 not-found.tsx page.  not-found.tsx is in the root app folder it does not have access to the `locale` parameter that we do in pages nested under `[locale]`

```bash
Error occurred prerendering page "/_not-found". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Could not find locale while pre-rendering page, make sure you called `setStaticParamsLocale` at the top of your pages
    at /home/zach/typehero/apps/web/.next/server/chunks/4964.js:1:150920
    at /home/zach/typehero/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:108:5119
    at /home/zach/typehero/apps/web/.next/server/chunks/4964.js:1:152590
    at Generator.next (<anonymous>)
    at /home/zach/typehero/apps/web/.next/server/chunks/4964.js:1:150131
    at new Promise (<anonymous>)
    at __async (/home/zach/typehero/apps/web/.next/server/chunks/4964.js:1:149909)
    at /home/zach/typehero/apps/web/.next/server/chunks/4964.js:1:152554
    at NotFound (/home/zach/typehero/apps/web/.next/server/chunks/7386.js:1:10186)

```

Reply:
We might have to opt-out of static rendering for this page. `setStaticParamsLocale` is used when building to retrieve the correct locale, so it should be used in this page (even if we hardcode `"en"`)

Reply:
> We might have to opt-out of static rendering for this page. `setStaticParamsLocale` is used when building to retrieve the correct locale, so it should be used in this page (even if we hardcode `"en"`)

so we need to add this to the top?

```ts
export const dynamic = 'force-dynamic';
```

Reply:
Just tried and `pnpm dev` works fine for 404 pages, and `pnpm build` also succeed

Reply:
Excellent, I tested also, works great, just pushed change!